context/src/main/java/io/opentelemetry/context/Context.java

```diff
@@ -138,6 +138,24 @@ static ExecutorService taskWrapping(ExecutorService executorService) {
     return new CurrentContextExecutorService(executorService);
   }
 
+  /**
+   * Returns an {@link ScheduledExecutorService} which delegates to the provided {@code
+   * executorService}, wrapping all invocations of {@link ExecutorService} methods such as {@link
+   * ExecutorService#execute(Runnable)} or {@link ExecutorService#submit(Runnable)} with the
+   * {@linkplain Context#current() current context} at the time of invocation.
+   *
+   * <p>This is generally used to create an {@link ScheduledExecutorService} which will forward the
+   * {@link Context} during an invocation to another thread. For example, you may use something like
+   * {@code ScheduledExecutorService dbExecutor = Context.wrapTasks(threadPool)} to ensure calls
+   * like {@code dbExecutor.execute(() -> database.query())} have {@link Context} available on the
+   * thread executing database queries.
+   *
+   * @since 1.43.0
+   */
+  static ScheduledExecutorService taskWrapping(ScheduledExecutorService executorService) {
```

Comment:
There's already a `Context.taskWrapping(ExecutorService)` method. `ScheduledExecutorService` extends `ExecutorService`, adding the following methods:
- `schedule(Runnable command, long delay, TimeUnit unit)`
- `schedule(Callable<V> callable, long delay, TimeUnit unit)`
- `scheduleAtFixedRate(Runnable command, long initialDelay, long period, TimeUnit unit)`
- `scheduleWithFixedDelay(Runnable command, long initialDelay, long delay, TimeUnit unit)`

I don't think it makes sense to propagate context for the scheduleAtFixedRate, scheduleWithFixedDelay methods.

I can see it being useful to propagate context via the schedule methods.